### PR TITLE
Fix TaskCache on net452

### PR DESCRIPTION
**Pull request by cesarbs**
To make https://github.com/aspnet/Identity/pull/1111 build without the hack in the .csproj.

### CODE REVIEW

File: shared/Microsoft.Extensions.TaskCache.Sources/TaskCache.cs [hunk 1]
```diff
@@ -14,10 +14,10 @@ internal static class TaskCache
         /// We're caching this in a static readonly field to make it more inlinable and avoid the volatile lookup done
         /// by <c>Task.CompletedTask</c>.
         /// </remarks>
-#if NET451
+#if NET451 || NET452
```
Comment (cesarbs) [inline shared/Microsoft.Extensions.TaskCache.Sources/TaskCache.cs]: I don't really like it, but I want to get things going.

Why don't we just do `Task.FromResult(0)` for both targets?

Comment (pakrym) [inline shared/Microsoft.Extensions.TaskCache.Sources/TaskCache.cs]: you can do #if !NETSTANDARD*, so each assembly would not have own completed task instance at least on core.

Comment (natemcmaster) [inline shared/Microsoft.Extensions.TaskCache.Sources/TaskCache.cs]: cc @rynowak @pakrym  do you remember why we use `Task.FromResult(0)` on desktop and `Task.CompletedTask` on CoreCLR?

Comment (Tratcher) [inline shared/Microsoft.Extensions.TaskCache.Sources/TaskCache.cs]: Because Task.CompletedTask is the one we want to use long term, but it's new in 4.6.

**Pull request merged**